src/main/java/seedu/address/model/transaction/TransactionList.java

```diff
@@ -23,6 +23,17 @@
     private final ObservableList<T> internalUnmodifiableList =
             FXCollections.unmodifiableObservableList(internalList);
 
+    /**
+     * Returns the total amount in a {@code transactionList}
+     */
+    public static <T extends Transaction> Amount getTotal(TransactionList<T> transactionList) {
```

Comment:
I think it can be non-static and just return the total amount obtained from the internal list (more straightforward and I think better practice than using a static method)